failureinjection: add disk stall cycle option

DarrylWong opened the pull request:
Some usages of disk stalls wish to repeatedly stall and unstall a disk in a short period of time. The current implementation requires an ssh connection per stall/unstall call, which can easily lead to flakes. e.g. the unstall call runs into an ssh flake and the node fatals before it can be recovered.

This change adds a disk stall cycle option which will do this in one single ssh connection until Recover is called.

Fixes: https://github.com/cockroachdb/cockroach/issues/148345
Release note: none

DarrylWong committed:
failureinjection: add disk stall cycle option

Some usages of disk stalls wish to repeatedly stall
and unstall a disk in a short period of time. The current
implementation requires an ssh connection per stall/unstall
call, which can easily lead to flakes. e.g. the unstall
call runs into an ssh flake and the node fatals before
it can be recovered.

This change adds a disk stall cycle option which will
do this in one single ssh connection until Recover is
called.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148704)
<!-- Reviewable:end -->

DarrylWong committed:
roachtestutil: disk stall Unstall returns error

Depending on the context of the test, Unstall erroring out
should result in a test failure. Some tests purposely induce
a node panic, in which case an error is expected and can be
safely ignored. However, some tests want to test that WAL
Failover works and any node death is unexpected.

This change modifies the disk stall API to return an error
for Unstall, and lets the test writer decide how to handle it.

DarrylWong commented:
Passed locally a few times but kicked off a TC run here with `count=5` for `disk-stalled/wal-failover/among-stores/with-progress`

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20026488?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true

xinhaoz reviewed (approved):
Nice, thanks for working on this!

DarrylWong commented:
TFTR!

bors r=xinhaoz
